src/compiler/program.ts

```diff
@@ -430,13 +430,14 @@ namespace ts {
         return program;
 
         function getCommonSourceDirectory() {
-            if (typeof commonSourceDirectory === "undefined") {
-                if (options.rootDir && checkSourceFilesBelongToPath(files, options.rootDir)) {
+            if (commonSourceDirectory === undefined) {

```

Comment:
I think test case is ok, and realized that scenario I was talking about sourceroot and maproot works because  --out is only allowed with ```amd``` or ```system``` and hence there is no way to get bundled emit that includes modules from ```node_modules``` folder.